File: Python/functions/no checking/perfect.py
#Perfect No.
def factor(n):
    s=0
    for i in range(1,n):
        if n%i==0:
            s+=i
    return s
n=int(input("Enter the no. = "))
if n==factor(n):
   print("Perfect No.") 
else:
    print("Not Perfect no.")

File: C Programms/No. Cheking In Range/perfect.c
//WAP to print Perfect no. for 1-n
#include<stdio.h>
void main()
{
    int n,r,s,x,i;
    printf("Enter the range = ");
    scanf("%d",&r);
    printf("Perfect No. from 1 to %d are ...\n",r);
    for(n=1;n<=r;n++)
    {
        for(i=1,s=0;i<n;i++)
        {
            if (n%i == 0)
                s+=i;            
        }
        if(s==n)
            printf("%d ",n);
    }
}